### PR TITLE
Refactor logic in mutate API into method

### PR DESCRIPTION
**Pull request by emilyreff7**
### Changes Included
Straightforward move of some pretty lengthy mutate logic into a reusable public method.

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -4273,7 +4217,72 @@ def mutate(table, exprs=None, **mutations):
         ] + non_overwriting_exprs
     else:
         proj_exprs = [table] + exprs
-    return table.projection(proj_exprs)
+    return proj_exprs
+
+
+def mutate(table, exprs=None, **mutations):
```
Comment (jreback) [inline ibis/expr/api.py:4223]: can you type as much as possible

**Review by jreback — CHANGES_REQUESTED** [review]
small request

File: ibis/expr/api.py (hunk 1 above)
Comment (emilyreff7) [reply ibis/expr/api.py]: done!

File: ibis/expr/api.py [hunk 2]
```diff
@@ -4144,67 +4144,11 @@ def _safe_get_name(expr):
...
-    """
-    exprs = [] if exprs is None else util.promote_list(exprs)
-    exprs.extend(
-        (expr(table) if util.is_function(expr) else as_value_expr(expr)).name(
-            name
-        )
-        for name, expr in sorted(mutations.items(), key=operator.itemgetter(0))
-    )
-
+def get_mutation_exprs(
```
Comment (jreback) [inline ibis/expr/api.py:4147]: can you move to ibis/expr/analysis (I think thats a good place for this; if you have a better idea all ears)

Comment (emilyreff7) [reply ibis/expr/api.py:4147]: Yeah that seems like a good place for it, will move!

**Review by jreback — APPROVED** [review]
lgtm ping on green

**Review by icexelloss — APPROVED** [review]
LGTM

**Pull request merged**